src/features/profile/index.tsx

```diff
@@ -0,0 +1,355 @@
+/* eslint-disable */
+import React, { Component } from 'react'
+import styles from './Profile.module.css'
+import classnames from 'classnames'
+import Rate from 'rc-rate'
+import 'rc-rate/assets/index.css'
+import '../../legacy/assets/css/rc-rate.custom.css'
+
+export default class Profile extends Component<any, any> {
+  render() {
+    // const {  } = this.props
+    return (
+      <div
+        className={classnames(
+          'px-8 py-10 w-full flex h-screen overflow-y-auto',
+          styles.profile,
+        )}
+      >
+        <div
+          className={classnames(
+            'm-auto w-full bg-white px-14 py-8 flex',
+            styles.wrapper,
+          )}
+        >
+          <div className='flex flex-col justify-between'>
+            <Card />
+            <div className='text-gray-400 text-sm'>
+              Member Since May 15, 2021
+            </div>
+          </div>
+          <div className='flex flex-col flex-grow pl-4'>
+            <Tabs />
+            <div className='flex pt-4 pl-2 justify-between flex-col lg:flex-col xl:flex-row'>
+              <General />
+              <Relations />
+            </div>
+          </div>
+        </div>
+      </div>
+    )
+  }
+}
+
+function truncateMiddle(
+  str: String,
+  frontLen: any,
+  backLen: any,
+  truncateStr: String,
+) {
+  if (str === null) {
+    return ''
+  }
+  var strLen = str.length
+  // Setting default values
+  frontLen = ~~frontLen // will cast to integer
+  backLen = ~~backLen
+  truncateStr = truncateStr || '&hellip;'
+  if (
+    (frontLen === 0 && backLen === 0) ||
+    frontLen >= strLen ||
+    backLen >= strLen ||
+    frontLen + backLen >= strLen
+  ) {
+    return str
+  } else if (backLen === 0) {
+    return str.slice(0, frontLen) + truncateStr
+  } else {
+    return str.slice(0, frontLen) + truncateStr + str.slice(strLen - backLen)
+  }
```

Comment:
I think we don't need to use `else` here.  It just need: 

`return str.slice(0, frontLen) + truncateStr + str.slice(strLen - backLen)`